src/vs/workbench/parts/files/electron-browser/fileActions.ts

```diff
@@ -160,8 +160,15 @@ class TriggerRenameFileAction extends BaseFileAction {
 		this._updateEnablement();
 	}
 
-
 	public validateFileName(name: string): string {
+		const names: string[] = name.split(/[\\/]/).filter(part => !!part);
+		if (names.length > 1) {	// error only occurs on multi-path
+			const comparer = isLinux ? strings.compare : strings.compareIgnoreCase;
+			if (comparer(names[0], this.element.name) === 0) {
+				return nls.localize('renameWhenSourcePathIsParentOfTargetError', "Cannot move/copy when source path is parent of target path.");
```

Comment:
I would improve this message to somehow be more clear and user friendly.
Also "move/copy" I prefer "move or copy"